### ISSUE TITLE
Start date is not reactive

### ISSUE THREAD

**Issue by MariusTechweb** [kenhyuwa/litepie-datepicker#31]
Hello!

I have a computed property for "startDate". I expect when this computed propety changes, then the datepicker will react to that and change the start position also.

**Comment by kenhyuwa** [comment]
hey, computed by default is getters only, you can use setter or using watcher

**Comment by MariusTechweb** [comment]
Hello, thanks for reply, could you maybe provide an example? @kenhyuwa 

When my calendar first loads, it's start date is today (correct), however, after some async operation, I want to change the start date to some other date.

**Comment by kenhyuwa** [comment]
```vue
methods: {
 someOperation(){
  this.calendar = [] // asigne with new date
 }
}
```

**Comment by MariusTechweb** [comment]
Hello, yes, I use this (see code below). However, when the calendar updates, and the computed property "`startDate`" updates, it is not visually shown in the `LitepieDatepicker` component. It still shows today as `startDate` despite it now being equal to the first date in the calendar array.
```
/**
 * Called when the component mounts.
 */
async mounted() {
  this.loading = true;
  this.calendar = await fetchCalendar(this.product.id);
  this.loading = false;

  console.log('loaded calendar for product ' + `${this.product.id} (${this.product.type})`, this.calendar);
},
```

**Comment by kenhyuwa** [comment]
are you value of calendar state match with your formatter?

**Comment by MariusTechweb** [comment]
Maybe not? I had to do a workaround for it to display the format as I want. Not sure if I bugged it? It does however should the date as I want.
```
<template>
  <LitepieDatepicker
    v-bind="pickerAttrs"
    :overlay="true"
    :shortcuts="false"
    :placeholder="placeholder"
    :formatter="formatter"
    :separator="separator"
    :start-from="startFrom"
    :disable-date="disableDate"
    :model-value="modelValue"
    @update:model-value="(value) => $emit('update:modelValue', value)"
  >
    <div class="flex items-center justify-between px-3 py-3 rounded cursor-pointer gap-x-4" :class="{ 'opacity-75 pointer-events-none': loading }" v-bind="containerAttrs">

      <!-- Label -->
      <div class="flex items-center gap-x-3">
        <font-awesome-icon
          class="text-lg text-current"
          :icon="[ 'far', 'calendar-alt' ]"
        />
        <span>
          {{ displayValue || placeholder }}
        </span>
      </div>

      <!-- Loading -->
      <spinner class="w-5 h-5" v-if="loading" />

    </div>
  </LitepieDatepicker>
</template>
```

Computed:
```
/**
 * The display value.
 */
displayValue() {
  // Get date format and seperator.
  const format    = 'll';
  const separator = this.separator;

  // Format value.
  if (Array.isArray(this.modelValue)) {
    return this.modelValue.map((d) => this.$moment(d).format(format)).join(separator);
  } else {
    return this.$moment(this.modelValue).format(format);
  }
},
```

**Comment by MariusTechweb** [comment]
By the way, @kenhyuwa , I am talking about changing the month outlined in red in the picture below to the same month as the first date in my calendar.
![bilde](https://user-images.githubusercontent.com/77326245/124580749-bc219700-de50-11eb-9160-8bdd44cde1d9.png)

**Comment by MariusTechweb** [comment]
Is there no way we could bind to these values so we can update them programatically?
```
    const panel = reactive({
      previous: {
        calendar: true,
        month: false,
        year: false
      },
      next: {
        calendar: true,
        month: false,
        year: false
      }
    });
```

**Comment by kenhyuwa** [comment]
> By the way, @kenhyuwa , I am talking about changing the month outlined in red in the picture below to the same month as the first date in my calendar array. For example, if the first date is in december, when I open the date picker, it should show the december month without me having to scroll there to see the calendar dates.
> ![bilde](https://user-images.githubusercontent.com/77326245/124580749-bc219700-de50-11eb-9160-8bdd44cde1d9.png)

the panel is binding from dateValue with formatter state. please make sure the state of value is match with formater

**Comment by MariusTechweb** [comment]
The issue is that I want the value to be in ISO format (to match server API), but the value actually displayed to be "ll" (another format). This is pretty critical. @kenhyuwa

**Comment by kenhyuwa** [comment]
yeah i know, for now you can format the response API with formatter matching before initialize the date.
I think, I resolve for next release.
thank you

**Comment by MariusTechweb** [comment]
So there is no way I can change the startDate after it has already been set? @kenhyuwa

**Comment by afunnydev** [comment]
@MariusTechweb If I may help, it looks like your initial `startFrom` and the async `startFrom` are maybe not in the same format. From your example, `old = "Tue Jul 06 2021 11:23:25 GMT+0200"` and `new = "2021-12-07"`.

From the [source code](https://github.com/kenhyuwa/litepie-datepicker/blob/e1232d53414000dfdcc642945635fb616867f53a/src/litepie-datepicker.vue#L1493-L1494), looks like `startFrom` doesn't use any `formatter` and should then be a `Date` object or ISO 8601 string (as per [day.js](https://day.js.org/docs/en/parse/string) `parse` recommandations without formatter). You should format `this.dates[0]` appropriately for it to get picked up, as `2021-12-07` is not ISO 8601. Hope this helps!

**Comment by MariusTechweb** [comment]
@afunnydev Hello and thank you, however, it seems like your provided solution does not work.

```javascript
/**
 * The start date.
 */
startDate() {
  return this.$moment(this.dates[0] || new Date()).toDate();
},
```

The start date (startFrom in the date picker) is still always today, regardless if `this.dates[0]` changes and becomes present as 10th of December.

**Comment by santiagopoveda** [comment]
I can also reproduce this issue, and you can even see it live on the main page: https://litepie.com/#as-single-use-range.

Type a valid date, and the date picker will close. Even though the date is actually set, the UI doesn't update.

Using @afunnydev's suggestion works, and while the UI does update when re-opened, an error is thrown when you set the date (at least when using a single date):

```TypeError: Cannot read properties of null (reading 'getBoundingClientRect') at useVisibleViewport```
